databroker/core.py

```diff
@@ -366,26 +364,20 @@ def documents_to_xarray(*, start_doc, stop_doc, descriptor_docs,
                 keys = scoped_data_keys
             for key, scoped_key in keys.items():
                 field_metadata = data_keys[key]
-                # Verify the actual ndim by looking at the data.
-                ndim = numpy.asarray(config['data'][key]).ndim
-                dims = None
-                if 'dims' in field_metadata:
-                    # As of this writing no Devices report dimension names ('dims')
-                    # but they could in the future.
-                    reported_ndim = len(field_metadata['dims'])
-                    if reported_ndim == ndim:
-                        dims = tuple(field_metadata['dims'])
-                    else:
-                        # TODO Warn
-                        ...
-                if dims is None:
-                    # Construct the same default dimension names xarray would.
+                field_metadata = data_keys[key]
+                ndim = len(field_metadata['shape'])
```

Comment:
Should this line be moved under the `except KeyError` below as it's done in the code block above?

Reply:
Good eye, but no, in this case we use it in an additional place below, regardless of whether we hit the `KeyError` codepath or not.

Reply:
You are right, I didn't check the code below carefully.